### PR TITLE
SNOW-1659276 add multithreading tests to sp env

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1659276

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Make required tweaks to `integ/test_multithreading.py` to enable them to run in SP Precommit.

### CODE REVIEW

**Comment by sfc-gh-aalam** [comment]
https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1407/

**Review by sfc-gh-jrose — APPROVED** [review]

File: tests/integ/test_multithreading.py [hunk 1]
```diff
@@ -256,30 +277,28 @@ def put_and_get_file(upload_file_path, download_dir):
...
     try:
         with ThreadPoolExecutor(max_workers=10) as executor:
-            futures = [
+            for package in package_list:
                 executor.submit(threadsafe_session.add_packages, package)
-                for package in package_list
-            ]
 
-            for future in as_completed(futures):
-                future.result()
```
Comment (sfc-gh-aling) [inline tests/integ/test_multithreading.py:276]: we don't need to wait for the task to finish here?

Comment (sfc-gh-aalam) [reply tests/integ/test_multithreading.py:276]: I moved `.get_packages()` outside of `with ThreadPoolExecutor()...` context. I learned from you that we come out after finishing all tasks. So made the test cleaner

Comment (sfc-gh-aling) [reply tests/integ/test_multithreading.py:276]: aha I see. it's a shame I forgot this already :P

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**